stdlib/public/Concurrency/ExecutorAssertions.swift

```diff
@@ -46,7 +46,8 @@ extension SerialExecutor {
   ///           the line where this method was called.
   @available(SwiftStdlib 5.1, *)
   #if !$Embedded
-  @backDeployed(before: SwiftStdlib 5.9)
+  // Corresponds to SwiftStdlib 5.9 without visionOS
+  @backDeployed(before: macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0)
```

Comment:
Huh; why would we want to do this? Is it just to avoid exposing the implementation in the .swiftinterface?

(AIUI `@backDeployed(before: SwiftStdlib 5.9)` would have largely the same effect in practice, as the deployment target on visionOS is unlikely to be below 1.0.)

Reply:
`SwiftStdlib 5.9` now expands to include `visionOS 1.0`, so if the macro were used here, it would cause a `@backDeployed` attribute to be implicitly created for the `visionOS` platform. That's not appropriate though; this API is _not_ back deployed on that platform. I'm afraid this is a thematic limitation of the availability macros feature. They get expanded very literally and sometimes that comes into conflict with the intuition for the effect you want them to have in practice.

Reply:
I think `SwiftStdlib 5.9` should not list `visionOS 1.0` -- the 5.9 APIs are always available on visionOS, so they do not need to be explicitly restricted to a specific version number.